### ISSUE TITLE
hexo admin very slow and crashing

### ISSUE THREAD

**Issue by ritzton** [jaredly/hexo-admin#152]
In my blog I have about 300 post with about 10 images for each post. Those images are copied in the asset folder of each post following this documentation https://hexo.io/docs/asset-folders.html.
To show the images in my articles I am writing {% asset_img image.jpg %} in my posts. I don't know why each time I am trying to create a new post through hexo-admin, the admin is not responding and I am getting the message "504 Gateway Time-out" when I am trying to reload the page.
I would like to know if I am doing anything wrong and why I do have those problems.

**Comment by pirtleshell** [comment]
Sounds like you have much more data than the admin is capable of handling in its current state. We could solve this with a lazy loading scheme on the posts list.

It's been a long-time dream to add better post filtering to the posts list (#3), so this would be a good thing to add in a hopefully-impending posts list overhaul.

**Comment by mr-m0nst3r** [comment]
Hey fellows,

I'm experiencing the same problem with ritzton, checked the code for some time, but unluckily, I'm just know little about NodeJS, but I guess the changing may be easy.

Can't show you the code because I just code NodeJS, but, `client/api/rest.js` this file might be an easier way to do with:
![image](https://user-images.githubusercontent.com/38524240/49450456-fa2d2900-f817-11e8-8058-706eef455d12.png)

```
return {
    posts: () => get('/posts/list'), // what about adding a param here?
    post: (id, data) => {
      if (data) return post('/posts/' + id, data)
      return get('/posts/' + id)
    },

```

if `posts` returns just all the title and `excerpt` as defined in `https://github.com/hexojs/hexo/blob/master/lib/models/post.js`, this will probably save a lot of traffic which is the leading cause of the problem.

dear @pirtleshell , would you like to take a look? Just love your hexo-admin, it's excellent.

**Comment by pirtleshell** [comment]
yes, the api will need to be modified to accept some filtering options, and a UI will need to be built or a particular loading scheme implemented in the post list.

migrating to the post lists to only initially load a stub of the post is a great idea that would resolve a lot of the data problems. that's probably some low-hanging fruit we could implement first before specific filtering/lazy loads. great suggestion!

**Comment by mr-m0nst3r** [comment]
Hi @pirtleshell , pls let me know when it's implemented, looking forward
